File: Spoj/Cpp Solution/updateit.cpp
#include <cstdio>
#include<iostream>
#include <vector>
#include<cstring>
using namespace std;
int bit[10001];
int t,n,u,q,id,l,r,val;
inline void Read (int *a)
{
    char c = 0;
    while(c<33)
    c = fgetc(stdin);
    *a = 0;
    while(c>33)
    {
               *a = *a*10 + c - '0';
               c = fgetc(stdin);
    }
}
void update(int x , int val )
{
	while ( x <= n)
	{
		bit[x] += val;
		x += ( x & -x);
	}
}
int sum(int x )
{
	int t = 0;
	while ( x > 0)
	{
		t += bit[x];
		x -= ( x & -x);
	}
	return t;
}

int main() {
	Read(&t);
    while(t--)
    {   Read(&n);
        Read(&u);
        memset(bit,0,sizeof bit);
        for(int i=0;i<u;i++){
            Read(&l);
            Read(&r);
            Read(&val);
            l++;r++;
           update(l,val);
           update(r+1,-1*val);
        }
        Read(&q);
       for(int i=0;i<q;i++)
        {
            Read(&id);
            id++;
            cout<<sum(id)<<endl;
        }
    }
	return 0;
}

File: Spoj/Python Solution/updateit.py
def update(bit , r , v , n):
    while r <= n:
        bit[r] += v
        r += ( r & -r )
def query(bit , r , n):
    sum = 0
    while r > 0:
        sum += bit[r]
        r -= ( r & -r)
    return sum
t = int(raw_input())
for i in range(t):
    n , u = map(int , raw_input().split())   
    bit = []
    for i in range(n+1):
        bit.append(0)
    for i in range(u):
        l , r, q = map(int , raw_input().split())
        update(bit , l+1 , q , n )
        update(bit , r+2 , (-1)*q , n)
         
    qu = int(raw_input())
    for i in range(qu):
        x = int(raw_input())
        print query( bit , x+1 , n)
    
